internal/sign/helper.go

```diff
@@ -16,29 +17,45 @@ func NewSignerHelper() Helper {
 	return &helper{}
 }
 
-func (m *helper) GetRelevantSign(modSpec kmmv1beta1.ModuleSpec, km kmmv1beta1.KernelMapping) *kmmv1beta1.Sign {
-	if modSpec.ModuleLoader.Container.Sign == nil {
-		// km.Sign cannot be nil in case mod.Sign is nil, checked above
-		return km.Sign.DeepCopy()
-	}
+func (m *helper) replaceInParam(macros map[string]string, params ...string) []string {
 
-	if km.Sign == nil {
-		return modSpec.ModuleLoader.Container.Sign.DeepCopy()
+	var replacedStrings []string
+	for k, v := range macros {
+		template := "${" + k + "}"
```

Comment:
that sounded so simple.... 

I've created an internal/utils/replaceStrings.go that contains functions to a) parse the kernel into major/minor/bugfix fields, and b) uses envsubst to do a find and replace with the idea that it will be much easier for users if we have a single set of substitutions we use everywhere.  I have then moved this PR to use it.

In some ways this  code duplicates whats in module/kernelmapper.go but is trying to be generic so we can use it anywhere else that needs similar functionality.  kernelmapper should probably be moved to being wrappers around replaceStrings as well, but I think that would make this PR much bigger and more sprawling than is good,